plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/notifications/NotificationPersistantState.kt

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.notifications
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+
+@State(name = "notificationDismissals", storages = [Storage("aws.xml")])
+class NotificationDismissalState : PersistentStateComponent<NotificationDismissalConfiguration> {
+    private var state = NotificationDismissalConfiguration()
+
+    override fun getState(): NotificationDismissalConfiguration = state
+
+    override fun loadState(state: NotificationDismissalConfiguration) {
+        this.state = state
+    }
+
+    fun isDismissed(notificationId: String): Boolean =
+        state.dismissedNotificationIds.contains(notificationId)
+
+    fun dismissNotification(notificationId: String) {
+        state.dismissedNotificationIds.add(notificationId)
+    }
+
+    companion object {
+        fun getInstance(): NotificationDismissalState =
+            ApplicationManager.getApplication().getService(NotificationDismissalState::class.java)
+    }
+}
+
+data class NotificationDismissalConfiguration(
+    var dismissedNotificationIds: MutableSet<String> = mutableSetOf(),
+)
+
+@State(name = "notificationEtag", storages = [Storage("aws.xml")])
+class NotificationEtagState : PersistentStateComponent<NotificationEtagConfiguration> {
+    private var state = NotificationEtagConfiguration()
+
+    override fun getState(): NotificationEtagConfiguration = state
+
+    override fun loadState(state: NotificationEtagConfiguration) {
+        this.state = state
+    }
+
+    var etag: String?
+        get() = state.etag
+        set(value) {
+            state.etag = value
+        }
+
+    companion object {
+        fun getInstance(): NotificationEtagState =
+            ApplicationManager.getApplication().getService(NotificationEtagState::class.java)
```

Comment:
```suggestion
        fun getInstance(): NotificationEtagState =
            ApplicationManager.getApplication().getService<NotificationEtagState>()
```

Reply:
I'm getting a syntax error when I do this. I think this is for newer versions of Intellij SDK. I don't see any other usages of the bottom code in our repository, but I do see the ::class.java version.

Reply:
can you try using getInstance(): NotificationETagState = service()

Reply:
that works!